core/network/impl/synchronizer_impl.cpp

```diff
@@ -1406,4 +1408,73 @@ namespace kagome::network {
     ancestry_.clear();
     recent_requests_.clear();
   }
+
+  bool SynchronizerImpl::fetchHeaderBack(const primitives::BlockInfo& block,
+    bool isFinalized, CbResultVoid cb) {
+    const auto blockNumber = block.number;
+    BlocksRequest request{
+        .fields = BlockAttribute::HEADER,
+        .from = blockNumber,
+        .direction = Direction::DESCENDING,
+        .max = 1,
```

Comment:
```suggestion
        .max = std::nullopt,
```

Reply:
The method is called fetchHeaderBack, not headers. It's done not to fetch excessive blocks, but one after another, untill it's needed. Its possible to implement fetchHeadersBack to fetch as many blocks as possible, but it's not the scope of this PR in my opinion. I don't understand why if we need just blocks from 24 to 22, we should load from 24 to 0 and moreover save it. Even fetching one by one in this particaular solution leads to a couple extra blocks being fetched and stored, but anyway it's better than just fetch and store everything.